lib/hooks.js

```diff
@@ -96,7 +96,7 @@ async function init (options) {
       // for the `request` hook, merge and return options data
       if (hookName === 'request') {
         // merge results of returned hook data
-        const finalData = data.options
+        const finalData = Hoek.clone(data.options)
```

Comment:
I feel like blindly cloning the entire object could lead to some poor performance, especially if one of the options is a large object. The possible options included here would be [these](https://github.com/ExpediaGroup/service-client/blob/master/lib/client.js#L110-L121):
```js
let requestOptions = {
    queryParams,
    pathParams,
    baseUrl,
    headers,
    payload,
    redirects,
    timeout,
    connectTimeout,
    maxConnectRetry,
    agent,
    read,
    readOptions
}
```

Most of these could probably easily be cloned, but some of them should be "shallow" copied instead (e.g. `agent`). I'm unsure what should be done with `payload`...

Reply:
Welp, we could just leave things as-is. The docs do point out that in the request hook data will be merged with the options data so while arguably not expected it is documented: 
> This hook is special. Data returned here will be deep merged with the request options data that will be passed on to Wreck.

Reply:
True, it's documented. The side effect still seems odd to me.

How about shallow copying everything except `queryParams`, `pathParams`, and `headers` since those are probably the most typical things that would be modified in a hook? Modifying `payload` or `agent` could get messy.

```suggestion
        const finalData = data.options;
        Object.assign(finalData, {
            queryParams: {...finalData.queryParams},
            pathParams: {...finalData.pathParams},
            headers: {...finalData.headers}
        });
```

The alternative (below) would only shallow copy the big objects that we know about. A user could return _any_ object from a request hook. So picking out specific objects we want to copy in the above example would be better than the one below, imo.

```suggestion
        const finalData = Hoek.clone(data.options, {shallow: ['payload', 'agent']});
```

Reply:
OK makes sense I was thinking of leveraging that `shallow` option if we were going to move forward with this update. Thanks!